File: sdk/python/examples/quickstart.py
"""GNOSIVELA Python quickstart.

Assumes a control plane running on http://localhost:8080 (in-memory mode is
fine): `cd backend && go run ./cmd/gnosivela`.
"""

from gnosivela import Client
from gnosivela.types import EntityRef, KnowledgeAssertion, Value

c = Client("http://localhost:8080")

# 1. create an ontology from the semantic contract DSL
dsl = """ontology procurement.supplier @version 1.0
entity Supplier identifiedBy canonicalSupplierId
property riskScore: Decimal [0..1]
"""
result = c.ontology_create(dsl)
print("ontology:", result["ontology"]["namespace"], result["ontology"]["version"])

# 2. register a canonical entity and resolve a hint
ent = c.entity_save(EntityRef(namespace="mdm", canonical_id="S-1042", type="Supplier", aliases=["ACME"]))
print("entity:", ent["canonicalId"])

# 3. propose a sourced assertion
a = KnowledgeAssertion(
    assertion_id="ka:risk",
    subject=EntityRef(namespace="mdm", canonical_id="S-1042", type="Supplier"),
    predicate="risk:score",
    object=Value(type="number", number=0.82),
    source="RiskOffice",
    status="validated",
    confidence=0.9,
)
print("assertion:", c.assertion_propose(a)["assertionId"])

# 4. run the consistency report and surface conflicts
report = c.consistency_report("procurement.supplier")
print("consistency failures:", report["failures"])

# 5. contract-driven query through the semantic bridge
view = c.bridge_query("procurement.supplier", "ACME risk", "risk-officer", "onboarding")
print("bridge view assertions:", len(view.get("assertions", [])))

File: sdk/typescript/examples/quickstart.ts
// GNOSIVELA TypeScript quickstart.
// Run with: node examples/quickstart.ts  (Node >= 22.6, no build step)
import { Client } from "../src/client.ts";

const c = new Client("http://localhost:8080");

async function main() {
  const dsl = `ontology procurement.supplier @version 1.0
entity Supplier identifiedBy canonicalSupplierId
property riskScore: Decimal [0..1]
`;

  const created = await c.ontologyCreate(dsl);
  console.log("ontology:", created.ontology?.namespace, created.ontology?.version);

  const ent = await c.entitySave({ namespace: "mdm", canonicalId: "S-1042", type: "Supplier", aliases: ["ACME"] });
  console.log("entity:", ent.canonicalId);

  await c.assertionPropose({
    assertionId: "ka:risk",
    subject: { namespace: "mdm", canonicalId: "S-1042", type: "Supplier" },
    predicate: "risk:score",
    object: { type: "number", number: 0.82 },
    source: "RiskOffice",
    status: "validated",
    confidence: 0.9,
  });

  const report = await c.consistencyReport("procurement.supplier");
  console.log("consistency failures:", report.failures);

  const conflicts = await c.consistencyConflicts();
  console.log("conflicts:", conflicts.conflicts.length);

  const view = await c.bridgeQuery("procurement.supplier", "ACME risk", "risk-officer", "onboarding");
  console.log("bridge view assertions:", (view.assertions as unknown[] | undefined)?.length ?? 0);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
